binary-quantization-mistral/bq-mistral-embed.md

```diff
@@ -0,0 +1,537 @@
+
+```python
+client.update_collection(
+    collection_name=f"{collection_name}",
+    optimizer_config=models.OptimizersConfigDiff(indexing_threshold=20000),
+)
+```
+
+### Advanced Search Options with Qdrant
+
+Qdrant is a vector search engine designed for efficient similarity search in large datasets. It is particularly useful for machine learning engineers and data scientists who need to manage and query embeddings or high-dimensional vectors. One of the key features of Qdrant is its ability to perform both approximate and exact searches, depending on the requirements of the task.
+
+### Performing a Search in Qdrant
+
+Using Qdrant to conduct a search involves specifying both the collection to search within and the query vector. The collection contains the indexed vectors, while the query vector represents the item you are searching for. The goal is to find the most similar vectors to the query vector within the collection.
+
+The following code snippet demonstrates how to perform a search in Qdrant:
+
+```python
+client.search(
+    collection_name=f"{collection_name}",
+    query_vector=points[32].vector,
+    search_params=models.SearchParams(
+        quantization=models.QuantizationSearchParams(
+            ignore=False,
+            rescore=False,
+            oversampling=2.0,
+        ),
+        exact=True,
+    ),
+    limit=5,
+)
+```
+
+This example uses the `search` method of a Qdrant client. The method takes several parameters, including the name of the collection to search (`collection_name`), the query vector (`query_vector`), search parameters (`search_params`), and a limit on the number of results (`limit`).
+
+### Understanding Search Parameters
+
+The search parameters play a crucial role in defining how the search is conducted. Here, `models.SearchParams` is used to specify the parameters, which include:
+
+- `quantization`: This parameter involves the settings for quantization, which can speed up the search process by approximating vector distances. The `oversampling` set to `2.0` indicates how much the quantization grid is denser than the minimum requirement. Setting `ignore` to `False` and `rescore` to `False` implies that quantization is used without ignoring it or rescoring the results post-quantization.
+- `exact`: Setting this to `True` forces an exact search instead of an approximate one, ensuring that the search results are as accurate as possible, albeit potentially at the expense of search speed.
+
+### Significance of Code Outputs
+
+The output of the search operation is represented by an empty list `['']`, indicating that no results were returned for the given query vector within the specified collection. This could result from various factors such as the vector not closely matching any in the collection, the search being too restrictive, or the collection not being properly populated.
```

Comment:
it does not make a lot of sense

it does not closely matching any in the collection - it requires setting a score threshold which had not been mentioned (I doubt we need to mention it here)